src/lightning_classify/models/model_builder.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any
+
+import torch
+from torch import nn
+from torchvision import models
+from torch.optim import Adam
+import lightning as L
+from torchmetrics import Accuracy
+
+
+class ImageRecogModel(L.LightningModule):
+    def __init__(self, lrate: float, num_classes: int = 18) -> Any:
+        super().__init__()
```

Comment:
Use save hyperparameter inside the __init__ to save every parameter inside the class

```python
  self.save_hyperparameters()
```

this is the example of how to use it :
```python
class LitMNIST(L.LightningModule):
    def __init__(self, layer_1_dim=128, learning_rate=1e-2):
        super().__init__()
        # call this to save (layer_1_dim=128, learning_rate=1e-4) to the checkpoint
        self.save_hyperparameters()

        # equivalent
        self.save_hyperparameters("layer_1_dim", "learning_rate")

        # Now possible to access layer_1_dim from hparams
        self.hparams.layer_1_dim
```
this is the URL for the example : https://lightning.ai/docs/pytorch/stable/common/lightning_module.html